tools/umi_tools/test/main.nf

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env nextflow
+
+// Define DSL2
+nextflow.preview.dsl=2
+
+// Log
+log.info ("Starting test for umi_tools dedup...")
+
+/*------------------------------------------------------------------------------------*/
+/* Define params
+--------------------------------------------------------------------------------------*/
+
+params.umitools_dedup_args = '--umi-separator=":"'
+params.verbose = true
+
+/*------------------------------------------------------------------------------------*/
+/* Module inclusions
+--------------------------------------------------------------------------------------*/
+
+include umitools_dedup from '../main.nf'
+
+/*------------------------------------------------------------------------------------*/
+/* Define input channels
+--------------------------------------------------------------------------------------*/
+
+// Meta data
+testMetaDataBam = [
+  ['sample1', "$baseDir/input/sample1.bam"],
+  ['sample2', "$baseDir/input/sample2.bam"],
+  ['sample3', "$baseDir/input/sample3.bam"],
+  ['sample4', "$baseDir/input/sample4.bam"],
+  ['sample5', "$baseDir/input/sample5.bam"],
+  ['sample6', "$baseDir/input/sample6.bam"]
+]
+
+testMetaDataBai = [
+  ['sample1', "$baseDir/input/sample1.bai"],
+  ['sample2', "$baseDir/input/sample2.bai"],
+  ['sample3', "$baseDir/input/sample3.bai"],
+  ['sample4', "$baseDir/input/sample4.bai"],
+  ['sample5', "$baseDir/input/sample5.bai"],
+  ['sample6', "$baseDir/input/sample6.bai"]
+]
+
+//Bam input channel
+Channel
+  .from(testMetaDataBam)
+  .map { row -> [ row[0], file(row[1], checkIfExists: true) ]}
+  .set {ch_test_meta_bam}
+
+//BamBai input channel
+Channel
+  .from(testMetaDataBai)
+  .map { row -> [ row[0], file(row[1], checkIfExists: true) ] }
+  .join(ch_test_meta_bam)
+  .set {ch_test_meta_bambai} 
+
+/*------------------------------------------------------------------------------------*/
+/* Run tests
+--------------------------------------------------------------------------------------*/
+
+workflow {
+    // Run dedup
+    umitools_dedup ( ch_test_meta_bambai )
+
+    // Collect file names and view output
+    umitools_dedup.out.dedupBam | view
```

Comment:
Is it possible to do some kind of assertion test against the results?

Reply:
We have been looking at a prototype using file hashes to verify output, or do you just mean verifying the file name outputs etc?

Reply:
Yeah file hashes is great too - just something to validate the contents of the file rather than just print it.

Reply:
This can get out of date quite quickly with updates to the main tool though?

Reply:
i.e. the file hash will fail because the output is different.

Reply:
Yes, you'd need to update it when you update the tool. Is that such a bad thing though?